### PR TITLE
Remove Stripe keys from the DB when uninstalling the plugin

### LINKED ISSUE #3384: Remove Keys from settings after uninstalling the plugin

### PR DESCRIPTION
**Pull request by diegocurbelo**
Fixes #3384

## Changes proposed in this Pull Request:

This PR updates the `uninstall.php` to always remove the Stripe keys when uninstalling the plugin/

It also updates the list of other `options` to remove when the `WC_REMOVE_ALL_DATA` constant is true.

## Testing instructions

NOTE: You should test this on new local site (or a JN site), not your dev env; otherwise, the "Delete" option will remove your working copy.

1. Build a package from this branch, or use this zip: [woocommerce-gateway-stripe.zip](https://github.com/user-attachments/files/16969143/woocommerce-gateway-stripe.zip)
2. Go to the Plugin Settings page:
    `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings`
3. Connect your Stripe account
4. Go to the Plugins page:
    `/wp-admin/plugins.php`
5. `Deactivate` and then `Delete` the `WooCommerce Stripe Gateway` plugin
6. Check that the plugin settings are still in the DB, for example using `wp-cli`:
    You can list all settings with:
    `wp option get woocommerce_stripe_settings`
7. Check that the Stripe API Keys were removed from the DB, for example using `wp-cli`
    You can list all settings with:
    `wp option get woocommerce_stripe_settings`
    Or individually check particular keys with:
    `wp option get woocommerce_stripe_settings`
    Screenshot 2024-09-02 at 4 50 20 PM
    For live mode, the keys are: `secret_key`, `publishable_key`, and `refresh_token` (only if using a Stripe App Connection), and for test mode: `test_secret_key`, `test_publishable_key`, and `test_refresh_token` (only if using a Stripe App Connection)
8. Install the Stripe Plugin again (you can use the same zip package)
9. Add the constant `WC_REMOVE_ALL_DATA` into your `wp-config.php` file:
    `define( 'WC_REMOVE_ALL_DATA', true );`
10. Connect your Stripe account
11. Go to the Plugins page:
    `/wp-admin/plugins.php`
12. `Deactivate` and then `Delete` the `WooCommerce Stripe Gateway` plugin.
13. Check that all the plugin settings were removed from the DB, for example using `wp-cli`:
    `wp option get woocommerce_stripe_settings`
    Screenshot 2024-09-11 at 4 36 37 PM

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [x] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [x] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [x] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

File: uninstall.php [hunk 1]
```diff
@@ -1,44 +1,120 @@
...
-	exit;
-}
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || exit;
+
+// Exit if uninstall not called from WordPress.
+defined( 'WP_UNINSTALL_PLUGIN' ) || exit;
+
+// Remove OAuth access_token refresh scheduled job.
+wp_clear_scheduled_hook( 'wc_stripe_refresh_connection' );
```
Comment (james-allan) [inline uninstall.php:15]: Yeah good call. 💯

File: uninstall.php [hunk 2]
```diff
@@ -1,44 +1,120 @@
 <?php
-if ( ! defined( 'ABSPATH' ) ) {
-	exit; // Exit if accessed directly.
-}
+/**
+ * WooCommerce Stripe Gateway Uninstall
+ *
+ * @version  x.x.x
```
Comment (james-allan) [inline uninstall.php:5]: Not sure if we want to define this or if it will bumped once the release happens.

**Review by james-allan — APPROVED** [review]
Changes look good. 

Just noting that while testing, my `test_webhook_secret` value was still stored in the db, but I think that's just an issue with the zip in the PR description. I can see the `test_webhook_secret` is being removed in the current changes and retrying with a freshly generated zip worked.

With `WC_REMOVE_ALL_DATA` set to true all the data was deleted.

**BEFORE**
Screenshot 2024-09-12 at 4 23 13 PM

**AFTER**

Screenshot 2024-09-12 at 4 23 20 PM

File: uninstall.php (hunk 2 above)
Comment (diegocurbelo) [reply uninstall.php:5]: WooRelease replaces it with the version number during the build process

**Pull request merged**